src/iocore/eventsystem/P_IOBuffer.h

```diff
@@ -551,6 +551,7 @@ IOBufferReader::is_read_avail_more_than(int64_t size)
 TS_INLINE void
 IOBufferReader::consume(int64_t n)
 {
+  ink_assert(read_avail() >= n);
```

Comment:
It looks like this function should handle this case, a). handle it as error case or b). consume only `read_avail()`. This assertion is a good start 👍

Reply:
I'm open to suggestions, but I don't think there's a good way to handle this, not here in consume. There's not enough context. In production, we don't want to spend the time in the `read_avail()`. If needed the caller should do that as a part of fulfilling its contract for this API. The debug assert here seems like a reasonable way to catch issues in debug builds.